### ISSUE TITLE
False negative: `LI_LAZY_INIT_STATIC` not reported when the null check is written Yoda-style (`null == x`)

### ISSUE THREAD

**Issue by Chordrain** [spotbugs/spotbugs#4138]
### SpotBugs version

SpotBugs 4.9.8 (JDK 21, javac 21, source/target 21)

### Check name (bug pattern)

`LI_LAZY_INIT_STATIC` — Incorrect lazy initialization of static field

### Description

The detector recognizes the unsynchronized lazy-initialization idiom only when the guard is written as `field == null`. Rewriting the *same* guard in Yoda style `null == field` — a semantics-preserving, symmetric reference comparison — makes the warning disappear, even though the lazy-init logic, the field, and the runtime behavior are unchanged.

### Reproducer

**Before:**

```java
public static Object getFoo() {
    if (foo == null)                   // reported
        foo = new Object();
    return foo;
}
public static String[] getY() {
    if (y == null) {                   // reported
        y = new String[] { "a", "b", "c", "d", "e" };
    }
    return y;
}
```

**After:**

```java
public static Object getFoo() {
    if (foo == null)                   // not reported
        foo = new Object();
    return foo;
}
public static String[] getY() {
    if (y == null) {                   // not reported
        y = new String[] { "a", "b", "c", "d", "e" };
    }
    return y;
}
```

### Expected behavior

`LI_LAZY_INIT_STATIC` should fire identically on both operand orderings of the null check, since `field == null` and `null == field` are the same condition.

### Actual behavior

- BEFORE: **2** reports (`LazyInit.foo` in `getFoo()`, `LazyInit.y` in `getY()`).
- AFTER:  **0** reports.
